main.tf

```diff
@@ -454,7 +462,7 @@ resource "aws_ecs_service" "ignore_changes_task_definition_and_desired_count" {
   platform_version                   = var.launch_type == "FARGATE" ? var.platform_version : null
   scheduling_strategy                = var.launch_type == "FARGATE" ? "REPLICA" : var.scheduling_strategy
   enable_ecs_managed_tags            = var.enable_ecs_managed_tags
-  iam_role                           = local.enable_ecs_service_role ? coalesce(var.service_role_arn, join("", aws_iam_role.ecs_service.*.arn)) : null
+  iam_role                           = local.enable_ecs_service_role ? coalesce(var.service_role_arn, join("", aws_iam_role.ecs_service[*].arn)) : null
```

Comment:
don't forget to update the tags to the new label module instance as well on this resource. can't comment there, because you did not make a change there.